CONTRIBUTING.md

```diff
@@ -0,0 +1,188 @@
+
+```
+npm i -g pnpm # Install pnpm
+```
+
+The project uses a [monorepo](https://monorepo.tools/) structure. It is divided into 3 main folders:
+
+- `packages` - This folder contains all the packages used within the Million.js library
+- `test` - This folder contains all the tests written for the Million.js library
+- `website` - This folder contains the code for the website [Million.js](https://millionjs.org/)
+
+### `packages` folder
+
+The `packages` folder contains all the code related to the Million.js package. The package is built  using Typescript.
+
+The source code is divided into five main folders:
+
+* `compiler` - This folder contains all the code related to the Million.js React and Next.js compilers
+* `jsx-runtime` - This folder contains the code related to the runtime code for jsx.
+* `million` - This folder contains the code for the core Million.js package. The implementation of the optimized array rendering (`<For/>`), `block()` and virtual DOM for React are in here.
+* `next` - This folder contains the Million.js support package for Next.js
+* `react` - This folder contains the Million.js support package for React
+
+To run the project locally, you need to install the dependencies.
+
+```bash
+pnpm install # Install dependencies
+cd {whichever of the packages}
+pnpm {command} # Run the project
+```
+
+`test` folder
+
+This folder contains the code for the tests for the implemetation of core Million.js features like the "map-array" with ``<For/>, ``block() and so on. The test use the [Vitest Framework](https://vitest.dev/).
+
+The source code is divided into three main modules:
+
+- `block.test.tsx` - This file contains the code for the test written for the block
+- `map-array.test.tsx` - This file contains the code for the test written for the optimized array rendering that `<For/>`  offers.
+- `react-compiler.test.tsx` - This file contains the code for the test written for the compiler built for react Million.js package.
+
+To run the test locally:
+
+```bash
+pnpm install # Install dependencies
+pnpm run test # Run all tests
+```
+
+`website` folder
+
+This folder contains the code for the Million.js website built with [Nextra](https://nextra.site/) a framework that allows you to build static websites with [Next.js](https://nextjs.org/) and [MDX.](https://mdxjs.com/)
+
+The source code is divided into two main folders:
+
+- `pages` - This folder contains all the pages of the website.
+- `components` - This folder contains all the components of the website.
+- `styles` - This folder contains all the styles of the website
+
+It is  suggested you to read the [Nextra](https://nextra.site/) to understand the project structure.
+
+To run the website locally, you need to install the dependencies and run the development server:
+
+```bash
+pnpm install # Install dependencies
+pnpm run dev # Run the development server
+```
+
+
+## To Test Million.js While Contributing
```

Comment:
I would name this "Create a dev playground"

Reply:
Done